### ISSUE TITLE
Unused VM resource false positives in templates

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#312]
**Description of the issue**

Managed disks and network interfaces only reference a VM after deployment. As a result, checking for unused disk or network interface resources in templates generates a false positive.

**Expected behaviour**

Exclude `Azure.VM.DiskAttached` and `Azure.VM.NICAttached` when validating from a template.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.10.0-B2002023**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      6.2.4
PSEdition                      Core
GitCommitId                    6.2.4
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
